### PR TITLE
feature(scheduler): plugin_evaluation_total metric support preScore/score

### LINKED ISSUE #110643: metrics to see which plugins affect to scheduler's decisions in Filter/Score phase
### What would you like to be added?

metrics to see more details related to scheduler's decisions in Filter phase

### Why is this needed?

In some cases (e.g. new feature is added in scheduler and want to see if it works or not), we may want to see which filter plugin affected Pod's scheduling results and it would be nice if we can observe that in metrics.

Ref: https://github.com/kubernetes/enhancements/pull/3338#discussion_r890411398

/cc @alculquicondor 
/sig scheduling

### PR DESCRIPTION
**Pull request by AxeZhan**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
"plugin_evaluation_total" metric counts how many times the specific plugin affects the scheduling result while taking Skip status into consideration. Previously it only supports preFilter/filter. With this pr, we can support preScore/score as well.
#### Which issue(s) this PR fixes:

Fixes #110643 

#### Does this PR introduce a user-facing change?

```release-note
"plugin_evaluation_total" metric supports prescore/score extension point.
The metric doesn't get incremented when the prescore/score plugin has nothing to do with an incoming Pod.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/scheduler/framework/runtime/instrumented_plugins.go [hunk 1]
```diff
@@ -52,3 +52,35 @@ func (p *instrumentedPreFilterPlugin) PreFilter(ctx context.Context, state *fram
...
+	framework.ScorePlugin
+
+	metric compbasemetrics.CounterMetric
+}
+
+var _ framework.ScorePlugin = &instrumentedScorePlugin{}
+
+func (p *instrumentedScorePlugin) Score(ctx context.Context, state *framework.CycleState, pod *v1.Pod, nodeName string) (int64, *framework.Status) {
+	score, status := p.ScorePlugin.Score(ctx, state, pod, nodeName)
+	if !status.IsSkip() {
```
Comment (kerthcet) [inline pkg/scheduler/framework/runtime/instrumented_plugins.go:82]: If PreScore is skipped, we'll not run Score. You can refer to PreFilter & Filter.

Comment (AxeZhan) [reply pkg/scheduler/framework/runtime/instrumented_plugins.go:82]: Ah yes. We should inc here anyway, I totally forgot😂.

File: pkg/scheduler/framework/runtime/framework.go [hunk 2]
```diff
@@ -372,6 +372,19 @@ func NewFramework(r Registry, profile *config.KubeSchedulerProfile, stopCh <-cha
 		}
 	}
 
+	// Cache metric streams for prescore and score plugins.
```
Comment (sanposhiho) [inline pkg/scheduler/framework/runtime/framework.go:375]: Can we have a separate function for L361 - L387?
If we add more extension points support on PluginEvaluationTotal in the future, here will get longer and longer.

Comment (AxeZhan) [reply pkg/scheduler/framework/runtime/framework.go:380]: Hmmm, I separated a function "setInstrumentedPlugins" to do this, I can't think of a more cleaner way🤔

File: pkg/scheduler/framework/runtime/zs_test.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
+package runtime
```
Comment (sanposhiho) [inline pkg/scheduler/framework/runtime/zs_test.go:1]: mistake?

**Review by sanposhiho — APPROVED** [review]
/lgtm

**Comment by AxeZhan** [comment]
/ping @sanposhiho @kerthcet  
I think you two can approve now? :)

**Comment by sanposhiho** [comment]
We should avoid one person giving both /lgtm /approve basically. I'd like to leave /approve to @kerthcet or @Huang-Wei

**Review by kerthcet — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**